api/v1alpha1/openapi.yaml

```
@@ -2502,6 +2514,244 @@ components:
           type: string
           description: 'Status of the operation. One of: "Success" or "Failure". More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status'
       description: Status is a return value for calls that don't return other objects.
+    DeviceWorkloadStatus:
+      type: object
+      required:
+        - status
+      properties:
+        summary:
+          $ref: '#/components/schemas/DeviceWorkloadSummaryStatus'
+          description: 'Summary status of a device workloads.'
+        applications:
+          type: array
+          description: 'Status of device applications.'
+          items:
+            $ref: '#/components/schemas/DeviceApplicationStatus'
+    DeviceApplicationStatus:
+      type: object
+      required:
+        - id
+        - name
+        - status
+      properties:
+        id:
+          type: string
+          description: 'Unique identifier for the application.'
+        name:
+          type: string
+          description: 'Human readable name of the application.'
+        status:
+          $ref: '#/components/schemas/DeviceApplicationStatusType'
+          description: 'Status of the application.'
+        ready:
+          type: string
+          description: 'The number of containers which are ready in the application.'
+        restarts:
+          type: integer
+          description: 'Number of restarts observed for the application.'
+    DeviceWorkloadSummaryStatus:
+      type: object
+      required:
+        - status
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceWorkloadSummaryType'
+          description: 'Summary status of workloads.'
+        info:
+          type: string
+          description: 'Human readable information indicating details about the workloads summary status.'
+    DeviceWorkloadSummaryType:
+      enum:
+      - 'Healthy'
+      - 'Degraded'
+      - 'Error'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'DeviceWorkloadSummaryHealthy'
+      - 'DeviceWorkloadSummaryDegraded'
+      - 'DeviceWorkloadSummaryRunning'
```

```suggestion
      - 'DeviceWorkloadSummaryRunning'
```

Please, remove this line as it causes the next variables to be shifted, and the open-api generator produces wrong types:


```
export enum DeviceWorkloadSummaryType {
  DeviceWorkloadSummaryHealthy = 'Healthy',
  DeviceWorkloadSummaryDegraded = 'Degraded',
  DeviceWorkloadSummaryRunning = 'Error',
  DeviceWorkloadSummaryError = 'Unknown',
}

```

---

oops, fixed